jobs/build-charms/charms.py

```diff
@@ -681,19 +696,11 @@ def charm_build(self):
                 args += " --charm-file"
                 self.dst_path = str(Path(self.src_path) / f"{self.name}.charm")
             self.echo(f"Building with: charm build {args}")
-            try:
-                _CharmStore(self).build(*args.split(), _cwd=self.src_path)
-            except sh.ErrorReturnCode:
-                ret.ok = False
-
+            ret = _CharmStore(self).build(*args.split(), _cwd=self.src_path)
         else:
             args = f"-f {self.src_path}"
             self.echo(f"Building with: charmcraft build {args}")
-            try:
-                _CharmHub(self).build(*args.split(), _cwd=self.build.build_dir)
-            except sh.ErrorReturnCode:
-                ret.ok = False
-
+            ret = _CharmHub(self).build(*args.split(), _cwd=self.build.build_dir)
```

Comment:
is _CharmStore used to build legacy charms for both the charm store and the hub?